src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReaderOptions.cs

```diff
@@ -9,6 +9,7 @@ namespace System.IO.Pipelines
     public class StreamPipeReaderOptions
     {
         private const int DefaultBufferSize = 4096;
+        private const int DefaultMaxBufferSize = 4096 * 1024;
```

Comment:
4MB? That might be too big 😄. Lets do 2MB for now (matches the array pool max).

Reply:
Fixed.